### PR TITLE
SOC-753 && SOC-755 Tracking for New Login flow

### PR DESCRIPTION
**Pull request by bkoval**
This Pull Request gathers all the commits from SOC-753 and SOC-755, previous pull requests:
https://github.com/Wikia/mercury/pull/862
https://github.com/Wikia/mercury/pull/966

Tickets:
https://wikia-inc.atlassian.net/browse/SOC-753
https://wikia-inc.atlassian.net/browse/SOC-755

### CODE REVIEW

File: server/facets/auth/join.ts [hunk 1]
```diff
@@ -28,7 +30,9 @@ function get (request: Hapi.Request, reply: any): Hapi.Response {
 			hideFooter: true,
 			bodyClasses: 'splash auth-landing-page',
```
Comment (kvas-damian) [inline server/facets/auth/join.ts]: We should remove this line

File: server/facets/auth/join.ts [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+import localSettings = require('../../../config/localSettings');
 import authView = require('./authView');
-var deepExtend = require('deep-extend');
+import deepExtend = require('deep-extend');
 
 interface JoinViewContext extends authView.AuthViewContext {
 	loginRoute: string;
 	facebookConnectHref: string;
 	signupHref: string;
+	trackingConfig: any;
```
Comment (kvas-damian) [inline server/facets/auth/join.ts]: trackingConfig should be included in ```authView.AuthViewContext``` as required.

File: server/facets/auth/join.ts [hunk 3]
```diff
@@ -28,7 +30,9 @@ function get (request: Hapi.Request, reply: any): Hapi.Response {
 			hideFooter: true,
 			bodyClasses: 'splash auth-landing-page',
 			noScripts: true,
-			signupHref: authUtils.getSignupUrlFromRedirect(redirectUrl)
+			signupHref: authUtils.getSignupUrlFromRedirect(redirectUrl),
+			bodyClasses: 'splash join-page',
+			trackingConfig: localSettings.tracking
 		}
 	);
```
Comment (kvas-damian) [inline server/facets/auth/join.ts]: What do you think about renaming template line below?

File: server/views/_layouts/auth.hbs [hunk 4]
```diff
@@ -32,10 +34,17 @@
...
 		<!-- build:js /front/scripts/auth.js -->
+		<script src="/front/vendor/i18next/i18next.min.js"></script>
+		<script src="/front/vendor/jquery/dist/jquery.js"></script>
+		<script src="/front/scripts/baseline.js"></script>
+		<script src="/front/scripts/mercury.js"></script>
 		<script src="/front/scripts/auth.js"></script>
 		<!-- endbuild -->
+
+		<script>
+			M.prop('tracking', {{{json trackingConfig}}});
```
Comment (kvas-damian) [inline server/views/_layouts/auth.hbs]: We plan to remove dependency on ```mercury.js```. Maybe we could follow the same scheme as ```translations``` and ```language``` few lines above?

File: front/scripts/main/components/LoginIconComponent.ts [hunk 5]
```diff
@@ -9,6 +9,12 @@ App.LoginIconComponent = Em.Component.extend({
...
 		if (Mercury.wiki.enableNewAuth) {
+			M.track({
+				trackingMethod: 'ga',
+				action: M.trackActions.click,
+				category: 'user-login-mobile',
+				label: 'join-link'
+			});
 			window.location.href = '/join?redirect=' + encodeURIComponent(window.location.href);
 		} else {
 			window.location.href = '/Special:UserLogin';
```
Comment (kvas-damian) [inline front/scripts/main/components/LoginIconComponent.ts]: Please add separate tracking for old login flow.

File: server/views/_layouts/auth.hbs (hunk 4 above)
Comment (bkoval) [inline server/views/_layouts/auth.hbs]: This code is used internally by Mercury.js so it won't be possible without refactoring it. I will set up a ticket to do that.

Comment (kvas-damian) [inline server/views/_layouts/auth.hbs]: Makes sense

Comment (bkoval) [inline server/views/_layouts/auth.hbs]: https://wikia-inc.atlassian.net/browse/SOC-939

**Comment by kvas-damian** [comment]
LGTM

**Comment by rafalkalinski** [comment]
LGTM

**Pull request merged**